File: C_Programmes/2D_Array_Search.c
//Design a C program to perform searching in a 2 Dimensional array

#include<stdio.h>
void Search(int row, int col, int [row][col],int);

int main()
{
    int key = 0;
    int row = 0, col = 0;
    printf("Enter the number of rows of the array\n");
    scanf("%d",&row);
    printf("Enter the number of columns of the array\n");
    scanf("%d",&col);
    int Array[row][col];
    printf("Enter the array elements\n");
    for (int i = 0; i<row; i++)
    {
        for(int j = 0; j<col; j++)
        {
            scanf("%d",&Array[i][j]);
        }
    }
    printf("Enter search element\n");
    scanf("%d",&key);
    Search(row, col, Array, key);
    return 0;
}

void Search(int row, int col , int Array[row ][col],int key)
{
    for (int i = 0; i<row; i++)
    {
        for(int j = 0; j<col; j++)
        {
            if(key == Array[i][j])
            {
                printf("The element is found at position [%d %d] \n", i+1, j+1);
                return;
            }
        }
    }
    printf("The element is not found \n");
    return;
}

File: Python3_Programmes/2D_Array_Search.py
def Search(row, col, Array, key):
    for i in range(0,row):
        for j in range(0,col):
            if key==Array[i][j]:
                print(f"The element is found at position [ {i+1} {j+1} ]",end="\n")
                return
    print("The element was not found",end="\n")
    return
def main():
    row = int(input("Enter the number of rows in the Array\n"))
    col = int(input("Enter the number of columns in the Array\n"))
    Array = list()
    print("Enter the array elements",end="\n")
    for i in range(0,row):
        Array.append([])
        for j in range(0,col):
            Array[i].append(int(input()))
    key=int(input("Enter the search key element\n"))
    Search(row, col, Array, key)
                
if __name__=='__main__':
    main()